backend/migrations/README.md

```diff
@@ -0,0 +1,111 @@
+
+Upon start of GraphQL container, sqlalchemy ```declarative_base``` is used to create all tables with this function: 
+```Base.metadata.create_all(engine.engine)```. **The number of tables depends on the modules that are enabled in ```config.json``` (in the root of the project).**
+
+As the database is created from scratch, it has no current information about migration state, so, first we need to run database upgrade.
+After that alembic will be able to generate the further migrations locally.
+
+This command will apply all migrations, and syncronize the DB state with local alembic history of migrations.
+```bash
+make upgrade-db 
+```
+or
+```bash
+alembic -c backend/alembic.ini upgrade head
+```
+
+To upgrade database and generate alembic migration during development use:
+
+```bash
+make generate-migrations
+```
+or
+```bash
+alembic -c backend/alembic.ini upgrade head
+alembic -c backend/alembic.ini revision -m "_describe_changes_shortly" --autogenerate
+```
+Please, change the auto-generated filename postfix with the short description of the migration purpose.
+
+## What to know about autogenerated migrations:
+
+**Autogenerate will detect:**
+
+ - Table additions, removals. 
+ - Column additions, removals.
+ - Change of nullable status on columns.
+ - Basic changes in indexes and explicitly-named unique constraints
+ - Basic changes in foreign key constraints
+
+**Autogenerate can optionally detect:**
+
+ - Change of column type. This will occur by default unless the parameter `EnvironmentContext.configure.compare_type` is set to `False`. The default implementation will reliably detect major changes, such as between `Numeric` and `String`, as well as accommodate for the types generated by SQLAlchemy’s “generic” types such as `Boolean`. Arguments that are shared between both types, such as length and precision values, will also be compared. If either the metadata type or database type has additional arguments beyond that of the other type, these are not compared, such as if one numeric type featured a “scale” and other type did not, this would be seen as the backing database not supporting the value, or reporting on a default that the metadata did not specify.
+
+The type comparison logic is fully extensible as well; see [Comparing Types](https://alembic.sqlalchemy.org/en/latest/autogenerate.html#compare-types) for details.
+
+ - Change of server default. This will occur if you set the `EnvironmentContext.configure.compare_server_default` parameter to `True`, or to a custom callable function. This feature works well for simple cases but cannot always produce accurate results. The Postgresql backend will actually invoke the “detected” and “metadata” values against the database to determine equivalence. The feature is off by default so that it can be tested on the target schema first. Like type comparison, it can also be customized by passing a callable; see the function’s documentation for details.
+
+**Autogenerate can not detect:**
```

Comment:
Maybe we should add a comment on the modules that need to be imported explictly, just to make sure that clients don't trust alembic blindly

Reply:
What imports do you mean?